java/cuvs-java/src/test/java/com/nvidia/cuvs/CagraBuildAndSearchTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.cuvs;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.nvidia.cuvs.common.SearchResults;
+
+public class CagraBuildAndSearchTest {
+  
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  /**
+   * A basic test that checks the whole flow - from indexing to search.
+   * 
+   * @throws Throwable
+   */
+  @Test
+  public void testIndexingAndSearchingFlow() throws Throwable {
+
+    // Sample data and query
+    float[][] dataset = {{ 0.74021935f, 0.9209938f }, { 0.03902049f, 0.9689629f }, { 0.92514056f, 0.4463501f }, { 0.6673192f, 0.10993068f }};
```

Comment:
Hardcoding tests makes it very hard to debug and troubleshoot problems, and they also become very brittle because we've found changes in the compute architectures can introduce flakiness and downright failures from hardcoded tests. Please generate the data instead of hardcoding the values and use bfknn to validate.